Intermittent failure of pullRevisionsWithAttachments[0]

We've seen this test fail randomly a couple of times it needs investigating
>
com.cloudant.sync.replication.AttachmentsPullTest > pullRevisionsWithAttachments[0] FAILED
    java.lang.RuntimeException at AttachmentsPullTest.java:235
    Caused by: java.net.SocketException at AttachmentsPullTest.java:235

and
>com.cloudant.sync.replication.AttachmentsPullTest > pullRevisionsWithAttachments[0] FAILED
    com.cloudant.mazha.CouchException
       Caused by: java.io.IOException

--- a/.travis.yml
+++ b/.travis.yml
@@ -3,7 +3,7 @@ sudo: false
 script: "./gradlew check integrationTest"
 
 jdk:
-  - openjdk7
+  - oraclejdk8
 
 services:
   - couchdb

--- a/build.gradle
+++ b/build.gradle
@@ -87,6 +87,16 @@ subprojects {
         configurations.testCompile.each { File file -> println file.name }
     }
 
+    tasks.withType(Test) {
+        testLogging {
+            // Get full exception info for test failures
+            exceptionFormat = 'full'
+            showExceptions = true
+            showCauses = true
+            showStackTraces = true
+        }
+    }
+
     //findbugs
     apply plugin: 'findbugs'
     findbugs {

